### DISCUSSION TITLE
What's the best way to migrate a large app from React Navigation to expo-router?

### DISCUSSION THREAD

**Discussion by adrianso** [expo/expo#27096] [Archived: Questions]
I am currently working on a large Expo app using React Navigation, and would like to migrate to expo-router.

Is there a way for these two navigation libraries to co-exist in the same project, so that I can migrate to expo-router screen by screen? I can't find any examples of that online.

**Comment by tkrtmy** [comment]
I recently wrote an article about my experience migrating a relatively large-scale React Native application from React Navigation to Expo Router. Unfortunately, from my experience, there wasn't a straightforward way to fully migrate incrementally on a screen-by-screen basis. However, I did manage to implement a gradual approach as much as possible.

The article is written in Japanese, but I believe browser translation tools will help make it accessible.

You can find the detailed write-up on Zenn (a Japanese tech blog platform) here:
<https://zenn.dev/tkrtmy/articles/6713ed8a5537b6>

Just wanted to put this out there for your reference!

**Reply by adrianso** [reply]
Unfortunately, I think you are correct. There is no easy way to migrate from React Native to Expo Router screen by screen.
What I managed in the end, is to migrate the my app navigator by navigator. It's very messy however, and I won't really recommend it to anyone.  :)
